dashboard/app/controllers/api/v1/users_controller.rb

```diff
@@ -34,6 +34,17 @@ def post_ui_tip_dismissed
     render status: 200, json: {set: property_name}
   end
 
+  # POST /api/v1/users/<user_id>/accept_data_transfer_agreement
+  def accept_data_transfer_agreement
+    @user.properties[:data_transfer_agreement_accepted] = true
+    @user.properties[:data_transfer_agreement_accepted_at] = DateTime.now
+    @user.properties[:data_transfer_agreement_request_ip] = params[:data_transfer_agreement_request_ip]
+    @user.properties[:data_transfer_agreement_source] = params[:data_transfer_agreement_source]
+    @user.save
+
+    render status: 200, json: {updated_user_properties: @user.properties}
```

Comment:
It looks like we're not using the returned JSON body at all.  Maybe this endpoint should just return a 204 NO CONTENT on success. (`head :no_content`, I think)

Reply:
✅